app/vmselect/prometheus/prometheus.go

```diff
@@ -302,15 +303,15 @@ func ExportHandler(startTime time.Time, w http.ResponseWriter, r *http.Request)
 	format := r.FormValue("format")
 	maxRowsPerLine := int(fastfloat.ParseInt64BestEffort(r.FormValue("max_rows_per_line")))
 	reduceMemUsage := httputils.GetBool(r, "reduce_mem_usage")
-	if err := exportHandler(nil, w, cp, format, maxRowsPerLine, reduceMemUsage); err != nil {
+	if err := exportHandler(nil, w, cp, format, maxRowsPerLine, reduceMemUsage, false); err != nil {
 		return fmt.Errorf("error when exporting data on the time range (start=%d, end=%d): %w", cp.start, cp.end, err)
 	}
 	return nil
 }
 
 var exportDuration = metrics.NewSummary(`vm_request_duration_seconds{path="/api/v1/export"}`)
 
-func exportHandler(qt *querytracer.Tracer, w http.ResponseWriter, cp *commonParams, format string, maxRowsPerLine int, reduceMemUsage bool) error {
+func exportHandler(qt *querytracer.Tracer, w http.ResponseWriter, cp *commonParams, format string, maxRowsPerLine int, reduceMemUsage bool, dropStaleNaNs bool) error {
```

Comment:
I don't think that it's needed to add additional arg - `dropStaleNaNs`.

 Stale NaNs must be dropped if `format == "promapi"`. This format is used only by `/api/v1/query` API.

Reply:
It makes no sense. This format is not documented any where and it's not possible to use such response for metrics import.

Reply:
If format == "promapi, stale nans will be dropped now. Please take a look.